torch/distributed/device_mesh.py

```diff
@@ -366,17 +384,40 @@ def __getitem__(self, mesh_dim_name: str) -> "DeviceMesh":
                 >>> # of cross-host(dim 0), and within-host (dim 1).
                 >>> mesh = DeviceMesh(device_type="cuda", mesh=[[0, 1, 2, 3],[4, 5, 6, 7]])
             """
-            if self.mesh.ndim == 1:
-                if self.mesh_dim_names and mesh_dim_name == self.mesh_dim_names[0]:
+            if not self.mesh_dim_names:
+                raise RuntimeError("Cannot slice a DeviceMesh without mesh_dim_names.")
+
+            mesh_dim_names = (
+                (mesh_dim_names,) if isinstance(mesh_dim_names, str) else mesh_dim_names
+            )
+
+            error_msg = (
+                f"Invalid mesh_dim_name {mesh_dim_names} specified. "
+                f"Valid mesh_dim_names should be a contiguous subsequence of {self.mesh_dim_names}."
+            )
+
+            # When the dimension slicing out is equal to the mesh dimensions of the current DeviceMesh,
+            # we simply return self if the given slicing is valid.
+            if len(mesh_dim_names) == len(self.mesh_dim_names):
```

Comment:
why we need this length check?

I feel we can simply do

```
if mesh_dim_names == self.mesh_dim_names:
                    return self
```
as it would check both content and length?

Reply:
Oops. Thanks for catching it. Was changing the sanity check a few times and forgot to remove this.